src/frontend/js/components/CourseProductCourseRuns/index.spec.tsx

```diff
@@ -0,0 +1,372 @@
+import { faker } from '@helpscout/helix';
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import fetchMock from 'fetch-mock';
+import type { PropsWithChildren } from 'react';
+import { IntlProvider } from 'react-intl';
+import { QueryClientProvider } from 'react-query';
+import {
+  ContextFactory as mockContextFactory,
+  CourseFactory,
+  JoanieCourseRunFactory,
+  JoanieEnrollmentFactory,
+  OrderLiteFactory,
+} from 'utils/test/factories';
+import JoanieApiProvider from 'data/JoanieApiProvider';
+import { CourseCodeProvider } from 'data/CourseCodeProvider';
+import { Priority } from 'types';
+import type { Course, CourseRun, Enrollment, OrderLite } from 'types/Joanie';
+import createQueryClient from 'utils/react-query/createQueryClient';
+import { Deferred } from 'utils/test/deferred';
+import { CourseRunList, EnrollableCourseRunList, EnrolledCourseRun } from '.';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockContextFactory({
+    authentication: { backend: 'fonzie', endpoint: 'https://auth.test' },
+    joanie_backend: { endpoint: 'https://joanie.test' },
+  }).generate(),
+}));
+
+describe('CourseProductCourseRuns', () => {
+  const dateFormatter = Intl.DateTimeFormat('en', {
+    day: '2-digit',
+    month: 'short',
+    year: 'numeric',
+  });
+
+  afterEach(() => {
+    fetchMock.restore();
+    JoanieCourseRunFactory.afterGenerate((cr: CourseRun) => cr);
+  });
+
+  describe('CourseRunList', () => {
+    const Wrapper = ({ children }: PropsWithChildren<{}>) => (
+      <IntlProvider locale="en">{children}</IntlProvider>
+    );
+
+    it('renders a warning message when no course runs are provided', () => {
+      render(
+        <Wrapper>
+          <CourseRunList courseRuns={[]} />
+        </Wrapper>,
+      );
+
+      expect(screen.getByText('No session available for this course.'));
+    });
+
+    it('renders a list of course runs', () => {
+      const courseRuns: CourseRun[] = JoanieCourseRunFactory.generate(2);
+
+      render(
+        <Wrapper>
+          <CourseRunList courseRuns={courseRuns} />
+        </Wrapper>,
+      );
+
+      // It should render all course runs provided
+      expect(screen.getAllByRole('listitem')).toHaveLength(2);
+
+      // For each course run, it should render course and enrollment dates
+      courseRuns.forEach((courseRun: CourseRun) => {
+        // - Course run start date should be displayed
+        screen.getByText(dateFormatter.format(new Date(courseRun.start)), {
```

Comment:
sweet I didn't know the use of the selector param  
![image](https://user-images.githubusercontent.com/76939263/160398679-fd7ca562-c4a8-4d86-8cd3-8bf87c73dc07.png)